src/game/Entities/VehicleHandler.cpp

```diff
@@ -42,12 +42,20 @@ void WorldSession::HandleDismissControlledVehicle(WorldPacket& recvPacket)
 
     Unit* vehicle = (Unit*)transportInfo->GetTransport();
 
+    if (!vehicle)
+        return;
     // Something went wrong
     if (vehicleGuid != vehicle->GetObjectGuid())
         return;
+    if (vehicle->IsBoarded())
+    {
+        vehicle = static_cast<Unit*>(vehicle->GetTransportInfo()->GetTransport());
+    }
 
     // Remove Vehicle Control Aura
     vehicle->RemoveSpellsCausingAura(SPELL_AURA_CONTROL_VEHICLE, _player->GetObjectGuid());
+
+    vehicle->GetVehicleInfo()->UnBoard(_player, false);
```

Comment:
not sure what I was thinking here either. unboarding from the parent vehicle for some reason?

Reply:
unboarding works recursively for subvehicles, so it's probably fine and possibly cleans up misapplied boardings in parent vehicles